src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -639,6 +626,63 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, Diagnostic
             }
         }
 
+        private BoundExpression BindTupleExpression(TupleExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            var arguments = node.Arguments;
+            if (arguments.Count < 2)
+            {
+                // this should be a parse error already.
+                return BadExpression(node);
+            }
+
+            var boundArguments = ArrayBuilder<BoundExpression>.GetInstance(arguments.Count);
+            var elementTypes = ArrayBuilder<TypeSymbol>.GetInstance(arguments.Count);
+            ArrayBuilder<string> elementNames = null;
+
+            for (int i = 0, l = arguments.Count; i < l; i++)
+            {
+                var argumentSyntax = arguments[i];
+                var name = argumentSyntax.NameColon?.Name.Identifier.ValueText;
+
```

Comment:
For the tuple types that result from a tuple creation expression - Yes. 
Note that generally it would be an error to provide names just for some fields. The code here still tries to handle partial named scenario for error tolerance. I am not sure yet if that scenario is going to be ultimately reachable, but I want to err on the safe side. If any of the names are missing (for whatever reason) they would become "ItemX" since that seems the only sensible way to handle such cases if they happen as a result of error recovery or something like that.